vllm_spyre/platform.py

```diff
@@ -515,3 +523,216 @@ def get_max_output_tokens(self, prompt_len: int) -> int:
                 max_new_tokens = max(max_new_tokens, shape['new_tokens'])
 
         return max_new_tokens
+
+    @classmethod
+    def _handle_disable_compilation(cls, vllm_config: VllmConfig,
+                                    is_decoder: bool):
+        """
+        For decoder models we want to respect the `REQUIRE_PRECOMPILED_DECODERS`
+        environment variable, which disallows torch_sendnn from compiling new
+        graphs and only allows it to load pre-compiled graphs.
+        In order to do this, we must load up some config from the torch_sendnn
+        cache and check to make sure that the current vllm config matches,
+        otherwise the cached artifacts cannot be used.
+
+        For encoder models, we do not allow disabling compilation.
+        """
+        req_precompiled_decoder_env_var = "REQUIRE_PRECOMPILED_DECODERS"
+
+        disable_flag = os.getenv(req_precompiled_decoder_env_var, "").lower()
+        if disable_flag not in ("1", "true"):
+            return
+
+        # If this isn't a decoder model, re-enable compilation
+        if not is_decoder:
+            logger.info("Unsetting %s because %s is not a decoder model",
+                        DISABLE_COMPILATION_ENV_VAR,
+                        vllm_config.model_config.model)
+            os.environ[DISABLE_COMPILATION_ENV_VAR] = "false"
+            return
+
+        # If the user asked to disable compilation, then we need to enforce that
+        # they setup their cache
```

Comment:
With the `REQUIRE_PRECOMPILED_DECODERS` change, we could make checks on `DISABLE_COMPILATION=true` requiring cache enablement happen for both decoders and encoders (separate from the validation for `REQUIRE_PRECOMPILED_DECODERS`).